model/textparse/benchmark_test.go

```diff
@@ -0,0 +1,183 @@
+// For local iteration, feel free to adjust cases/comment out code etc.
+//
+// NOTE(bwplotka): Do not try to conclude "what parser (OM, proto, prom) is the fastest"
+// as the testdata has different amount and type of metrics and features (e.g. exemplars).
+func BenchmarkParse(b *testing.B) {
+	for _, bcase := range []struct {
+		dataFile  string // localized to ./testdata
+		dataProto []byte
+		parser    string
+
+		compareToExpfmtFormat expfmt.FormatType
+	}{
+		{dataFile: "promtestdata.txt", parser: "promtext", compareToExpfmtFormat: expfmt.TypeTextPlain},
+		{dataFile: "promtestdata.nometa.txt", parser: "promtext", compareToExpfmtFormat: expfmt.TypeTextPlain},
+
+		//  We don't pass compareToExpfmtFormat: expfmt.TypeProtoDelim as expfmt does not support GAUGE_HISTOGRAM:
+		// "expfmt.extractSamples: unknown metric family type GAUGE_HISTOGRAM"
+		// TODO(bwplotka): Fix expfmt?
+		{dataProto: createTestProtoBuf(b).Bytes(), parser: "promproto"},
+
+		// We don't pass compareToExpfmtFormat: expfmt.TypeOpenMetrics as expfmt does not support OM exemplars:
+		// text format parsing error in line 19: expected integer as timestamp, got "#"
+		// TODO(bwplotka): Fix expfmt?
+		{dataFile: "omtestdata.txt", parser: "omtext"},
+		{dataFile: "promtestdata.txt", parser: "omtext"}, // Compare how omtext parser deals with Prometheus text format vs promtext.
+	} {
+		var buf []byte
+		dataCase := bcase.dataFile
+		if len(bcase.dataProto) > 0 {
+			dataCase = "createTestProtoBuf()"
+			buf = bcase.dataProto
+		} else {
+			f, err := os.Open(filepath.Join("testdata", bcase.dataFile))
+			require.NoError(b, err)
+			b.Cleanup(func() {
+				_ = f.Close()
+			})
+			buf, err = io.ReadAll(f)
+			require.NoError(b, err)
+		}
+		b.Run(fmt.Sprintf("data=%v/parser=%v", dataCase, bcase.parser), func(b *testing.B) {
+			newParserFn := newTestParserFns[bcase.parser]
+			var (
+				res labels.Labels
+				e   exemplar.Exemplar
+			)
+
+			b.SetBytes(int64(len(buf)))
+			b.ReportAllocs()
+			b.ResetTimer()
+
+			st := labels.NewSymbolTable()
+			for i := 0; i < b.N; i++ {
+				p := newParserFn(buf, st)
+
+			Inner:
+				for {
+					t, err := p.Next()
+					switch t {
+					case EntryInvalid:
+						if errors.Is(err, io.EOF) {
+							break Inner
+						}
+						b.Fatal(err)
+					case EntryType:
+						_, _ = p.Type()
+						continue
+					case EntryHelp:
+						_, _ = p.Help()
+						continue
+					case EntryUnit:
+						_, _ = p.Unit()
+						continue
+					case EntryComment:
+						continue
+					case EntryHistogram:
+						_, _, _, _ = p.Histogram()
+					case EntrySeries:
+						_, _, _ = p.Series()
+					default:
+						b.Fatal("not implemented entry", t)
+					}
+
+					_ = p.Metric(&res)
+					_ = p.CreatedTimestamp()
+					for hasExemplar := p.Exemplar(&e); hasExemplar; hasExemplar = p.Exemplar(&e) {
+					}
+				}
+			}
+		})
+
+		// Compare with expfmt, opt-in, no need to benchmark external code.
```

Comment:
This comment is pretty confusing as by default we run this check for the two text parsers, so it's not opt-in.

Reply:
I forgot to revert commented skip, but I guess that's fine to do it opt-out, not a biggie. It's easy to decide on what to run with `-bench` regex.